src/wl_monitor.c

```diff
@@ -130,7 +136,8 @@ void _glfwAddOutputWayland(uint32_t name, uint32_t version)
         return;
     }
 
-    monitor = _glfwAllocMonitor(name_str, 0, 0);
+    // The actual name of this output will be set in the geometry handler.
+    monitor = _glfwAllocMonitor("wl_output", 0, 0);
```

Comment:
It's a bit awkward to have to pass a temporary value. Can't we just fix _glfwAllocMonitor() to only strdup if name is not NULL?